packages/components/timed-text-editor/WrapperBlock.js

```diff
@@ -80,22 +78,52 @@ class WrapperBlock extends React.Component {
     }
 
     if(nextProps.block.getData().get('speaker') !== this.state.speaker){
-      console.log('shouldComponentUpdate wrapper speaker', nextProps.block.getData().get('speaker') , this.state.speaker )
       return true;
     }
     return false;
   };
 
   componentDidUpdate  = (prevProps, prevState) =>{
-
-    if(prevProps.block.getData().get('speaker') !== prevState.speaker){
-        // console.log('componentDidUpdate wrapper speaker', prevProps.block.getData().get('speaker') , prevState.speaker );
+    const { block } = this.props;
+    if(block.getData().get('speaker') !== prevState.speaker){
         this.setState({
-          speaker: prevProps.block.getData().get('speaker')
+          speaker: block.getData().get('speaker'),
+          words: prevProps.block.getData().get('words') ? block.getData().get('words') : []
         })
 
         return true;
       }
+
+      if(block.getData().get('words') !== prevState.words){
+        const start = block.getData().get('start');
+
+        this.setState({
+          start: start,
+          words: this.getWordsBeforeBlock(block)
+        });
+
+        return true;
+      }
+  }
+
+  // we get all words inside the blocks before the current one
+  // and add them to the attribute data prev-time
+  // to do css injection to show current paragraph positioning in text
+  getWordsBeforeBlock = (block)=>{
+    const blockId = block.getKey();
+        const currentContent = this.props.blockProps.editorState.getCurrentContent();
+        const blocksAsArray = currentContent.getBlocksAsArray();
+        const currentBlockIndex = blocksAsArray.findIndex((block)=>{
+          return  blockId=== block.getKey()
+        }) 
+    
+        const blocksBeforeCurrentBlock = blocksAsArray.slice(0,currentBlockIndex);
+        const wordsBefore = blocksBeforeCurrentBlock.map((block)=>{
+          const words = block.getData().get('words') ? block.getData().get('words') : [];
```

Comment:
JFYI, an easier way to assign a default value if the first variable is not defined.
```suggestion
          const words = block.getData().get('words') || [];
```

Reply:
ok, fair